pkg/ldbwriter/ldb_writer.go

```diff
@@ -191,3 +191,37 @@ func (writer *SqlLdbWriter) Close() error {
 	}
 	return nil
 }
+
+// PragmaWALResult https://www.sqlite.org/pragma.html#pragma_wal_checkpoint
+type PragmaWALResult struct {
+	// 0 indicates not busy, checkpoint was not blocked from completing. 1 is blocked
+	Busy int
+	// number of modified pages that have been written to the write-ahead log file
+	Log int
+	// number of pages in the write-ahead log file that have been successfully moved back into the database file at the conclusion of the checkpoint
+	Checkpointed int
+}
+
+func (p *PragmaWALResult) String() string {
+	return fmt.Sprintf("busy=%d, log=%d, checkpointed=%d", p.Busy, p.Log, p.Checkpointed)
+}
+
+// QueryCheckpoint queries the wal checkpoint stats sqlite updates as it manages the WAL file
+// see https://www.sqlite.org/pragma.html#pragma_wal_checkpoint for more details
+// requires write access
+func (writer *SqlLdbWriter) QueryCheckpoint() (*PragmaWALResult, error) {
+	res, err := writer.Db.Query("PRAGMA wal_checkpoint")
```

Comment:
I did misunderstand this a bit, I did think calling it without a argument was equivalent to a stats query.  

But executing just `PRAGMA wal_checkpoint` is equivalent to checkpointing in passive mode, so I don't believe it'll interfere with writer or reader performance. 

I'm mainly interested in the busy response. Our theory is that our WALs are growing unbounded because the checkpointing can never complete due to open read connections. If this returns with 1 repeatedly, it indicates that checkpointing isn't able to make progress. 

It is per a checkpoint attempt:
```
sqlite> pragma wal_checkpoint;
busy        log         checkpointed
----------  ----------  ------------
0           68          68
sqlite> pragma wal_checkpoint;
busy        log         checkpointed
----------  ----------  ------------
0           5           5
sqlite> pragma wal_checkpoint;
busy        log         checkpointed
----------  ----------  ------------
0           5           5
sqlite> pragma wal_checkpoint;
busy        log         checkpointed
----------  ----------  ------------
0           5           5
sqlite> pragma wal_checkpoint;
busy        log         checkpointed
----------  ----------  ------------
0           5           5
sqlite> pragma wal_checkpoint;
busy        log         checkpointed
----------  ----------  ------------
0           10          10
sqlite> pragma wal_checkpoint;
busy        log         checkpointed
----------  ----------  ------------
0           10          10
sqlite> pragma wal_checkpoint;
busy        log         checkpointed
----------  ----------  ------------
0           10          10
sqlite> pragma wal_checkpoint;
busy        log         checkpointed
----------  ----------  ------------
0           11          11
sqlite> pragma wal_checkpoint;
busy        log         checkpointed
----------  ----------  ------------
0           5           5
sqlite> pragma wal_checkpoint;
busy        log         checkpointed
----------  ----------  ------------
0           10          10
```

I will update the docs here and also include configuration disabling this entirely.

Reply:
Given that the output seems to be per-run, and is in this case always identical, might it provide value to emit the raw log & checkpointed metrics as well?   Perhaps we can try that for now in the canarying of this in prod where we see issues.

Reply:
Thank you!  Excited to see the results of this.

Reply:
out of curiosity, is there specific reason why we choose `PASSIVE` mode over other modes, say `FULL` mode when performing the checkpointing?

Reply:
> out of curiosity, is there specific reason why we choose `PASSIVE` mode over other modes, say `FULL` mode when performing the checkpointing?

Passive is the default that the reflector is already performing. Since this is to gather metrics rather than change behavior, I didn't want to use a more aggressive checkpointing.